### PR TITLE
Test 3 february

### PR DESCRIPTION
**Pull request by Shohjahont9**

### CODE REVIEW

**Review by NikitaLychkovski — APPROVED** [review]

File: app/src/main/java/uz/larsLion/deliverytakeout/data/useCaseImp/GetKitchenCardUseCaseImp.kt [hunk 1]
```diff
@@ -3,13 +3,20 @@ package uz.larsLion.deliverytakeout.data.useCaseImp
...
 class GetKitchenCardUseCaseImp(
     private val repository: KitchenCardRepository
 ) : GetKitchenCardUseCase {
 
-    override suspend fun loadKitchenCardData(): List<KitchenCardItemDTO> {
+    override suspend fun loadKitchenCardData(date: String): List<KitchenCardItemDTO> {
         val allEvents = repository.getKitchenCards()
-        return allEvents.sortedBy { it.preparationTime }
+        return allEvents.sortedBy { it.preparationTime }.filter {
+            eeee_mmmm_dd.format(yyyy_mm_dd_hh_mm_ss_z.parse(it.pickupTime))==date
```
Comment (NikitaLychkovski) [inline app/src/main/java/uz/larsLion/deliverytakeout/data/useCaseImp/GetKitchenCardUseCaseImp.kt:18]: It is not a good idea to work with dates as strings please use Dates objects:
1. You can parse `pickupTime` into Date when you create KitchenCardItemDTO
2.  loadKitchenCardData(date: String) better to send here Date object as well

File: app/src/main/java/uz/larsLion/deliverytakeout/presentation/mainScreen/MainScreenViewModel.kt [hunk 2]
```diff
@@ -43,4 +49,48 @@ class MainScreenViewModel(
...
+
+    fun previousDate() {
+        dayCounter--
+        setDate(dayCounter)
+    }
+
+    fun defaultDate(){
+        dayCounter = 0
+        setDate(dayCounter)
+    }
```
Comment (NikitaLychkovski) [inline app/src/main/java/uz/larsLion/deliverytakeout/presentation/mainScreen/MainScreenViewModel.kt:81]: this is part of logic may be better to move it to viewModel

File: app/src/main/java/uz/larsLion/deliverytakeout/presentation/mainScreen/MainScreenViewModel.kt [hunk 3]
```diff
@@ -43,4 +49,48 @@ class MainScreenViewModel(
...
+        calendarPrev.add(Calendar.DAY_OF_YEAR, dayCounter - 1)
+        val calendarNext = Calendar.getInstance()
+        calendarNext.add(Calendar.DAY_OF_YEAR, dayCounter + 1)
+        val calendarCurrent = Calendar.getInstance()
+        calendarCurrent.add(Calendar.DAY_OF_YEAR, (dayCounter))
+        _getDate.value =
+            eeee_mmmm_dd.format(calendarPrev.time) + "|" + eeee_mmmm_dd.format(calendarCurrent.time) + "|" + eeee_mmmm_dd.format(
+                calendarNext.time
+            )
+    }
```
Comment (NikitaLychkovski) [inline app/src/main/java/uz/larsLion/deliverytakeout/presentation/mainScreen/MainScreenViewModel.kt:94]: you could create an object that contains all this dates and pass it. Multiple serialization just to pass params from viewModel to fragment is not good idea. Also, it not even clear from field name `date` to contain the string "date|date|date"

File: app/src/main/java/uz/larsLion/deliverytakeout/presentation/mainScreen/MainScreenViewModel.kt [hunk 4]
```diff
@@ -5,33 +5,39 @@ import androidx.lifecycle.MutableLiveData
...
 class MainScreenViewModel(
     private val useCase: GetKitchenCardUseCase
 ) : ViewModel() {
 
     private val getKitchenList = mutableListOf<KitchenCardItemDTO>()
+    var dayCounter = 0
+
 
     private val _getKitchenData = MutableLiveData<State<List<KitchenCardItemDTO>>>()
     val getKitchenData: LiveData<State<List<KitchenCardItemDTO>>> = _getKitchenData
```
Comment (NikitaLychkovski) [inline app/src/main/java/uz/larsLion/deliverytakeout/presentation/mainScreen/MainScreenViewModel.kt:24]: As I said before, your variables must NOT start with Verb. Only functions could start with Verb

File: app/src/main/java/uz/larsLion/deliverytakeout/presentation/mainScreen/MainScreenViewModel.kt [hunk 5]
```diff
@@ -5,33 +5,39 @@ import androidx.lifecycle.MutableLiveData
...
+        dayCounter = 0
     }
 
-    fun kitchenData() {
+    fun kitchenData(date: String) {
         if (getKitchenList.isNotEmpty()) {
             _getKitchenData.value = State.Content(getKitchenList)
         }
         _getKitchenData.value = State.Loading
         viewModelScope.launch {
```
Comment (NikitaLychkovski) [inline app/src/main/java/uz/larsLion/deliverytakeout/presentation/mainScreen/MainScreenViewModel.kt:36]: viewModelScope is running in main thread

**Comment by Shohjahont9** [comment]
fixing Coroutine codes, dates

**Pull request merged**